src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketSCMSource.java

```diff
@@ -666,7 +666,7 @@ private void retrievePullRequests(final BitbucketSCMSourceRequest request)
                                 repoOwner,
                                 repository,
                                 repositoryType,
-                                branchName,
+                                pull.getSource().getBranch().getName(),
```

Comment:
Maybe save this to a new variabile near branchName named originalBranchName just to remember in the future what is It and why take from pull instance instead use already avaiable branchName.